temporal-sdk/src/main/java/io/temporal/client/WorkflowClient.java

```diff
@@ -237,7 +239,29 @@ WorkflowStub newUntypedWorkflowStub(
    */
   WorkflowExecution signalWithStart(BatchRequest signalWithStartBatch);
 
-  WorkflowServiceStubs getWorkflowServiceStubs();
+  /**
+   * A wrapper around {WorkflowServiceStub#listWorkflowExecutions(ListWorkflowExecutionsRequest)}
+   *
+   * <p>Note: This method uses 1000 as the underlying page size. To customize the page size, use
+   * {@link #listWorkflowExecutions(String, int)}
+   *
+   * @param query Temporal Visibility Query, for syntax see <a
+   *     href="https://docs.temporal.io/visibility#list-filter">Visibility docs</>
+   * @return iterable that performs remote pagination under the hood
+   */
+  default Iterable<WorkflowExecutionMetadata> listWorkflowExecutions(String query) {
```

Comment:
Wonder if we should call this list `list` or `listExecutions`. We are already in a `WorkflowClient` and other service calls don't seem to include the extra `Workflow` term.

Reply:
I'm a little reserved about using Stream in public APIs of libraries if stream manipulation is not the sole purpose of these APIs. If the producer and consumer of the Stream are in the same codebase - sure, why not, it's handy.

Otherwise, Stream instance carries a bunch of configurations with it that the producer did. Specifically, Spliterator characteristics and if the Stream is parallel or not. These configurations is often an implementation detail of a CONSUMER and important for a CONSUMER, not a producer. Exposing Streams, we make consumers to
1. guess and expect that parameters are correct
2. assert that the returned Stream has expected characteristics
3. or recreate the Stream with the correct characteristics.


I think it's a better and unambiguous experience to return an Iterable and let the consumer create a Stream with the parameters that the consumer expects / relies on.

But I may see how users may no want to do that and expect a simpler higher-level API and if you have a strong opinion - I can rework on returning a Stream.

Reply:
I really prefer it to don't be just `list`. Just because there may be so many other stuff we may want to `list` that may be handy on WorkflowClient.

Belonging to `Workflow`Client is a bit of a weak argument, because one day we may want to list some other executions, that are a part of a workflow, but not a workflow itself, like an Activity.

But I'm fine with listExecutions if you prefer it.

Reply:
It just seems off to add `Workflow` in the high-level service wrapper when we don't for start/execute, signal with start, signal, query, etc. In Python we add it because we have a generic client that has workflow and async activity completion and other things in it, but here you're already scoped to a workflow. I can't personally imagine how any other use of `list` on a "workflow client" would make more sense than listing workflows, but I think `listExecutions` at least is in spirit with the other method names.